src/components/structures/ContextMenu.tsx

```diff
@@ -429,10 +429,10 @@ export type AboveLeftOf = IPosition & {
 // Placement method for <ContextMenu /> to position context menu right-aligned and flowing to the left of elementRect,
 // and either above or below: wherever there is more space (maybe this should be aboveOrBelowLeftOf?)
 export const aboveLeftOf = (
-    elementRect: DOMRect,
+    elementRect: Partial<DOMRect>,
```

Comment:
if we want a partial, we should define what properties we actually want instead (so that DOMRect automatically fulfills that interface).